File: exercise/leetcode/0128/0128.py
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
            
        s = set()
        for i, v in enumerate(nums):
            s.add(v)

        max_len = 1
        for i, v in enumerate(nums):
            if v-1 in s:
                continue
            t = v + 1
            cnt = 1
            while t in s:
                t += 1
                cnt += 1
            max_len = max(max_len, cnt)
        
        return max_len

File: exercise/leetcode/0128/0128.cc
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> us;
        for (int i = 0; i < nums.size(); ++i) {
            us.insert(nums[i]);
        }
        
        int max_len = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (us.find(nums[i]-1) == us.end()) {
                int cnt = 0;
                int v = nums[i];
                while (us.find(v) != us.end()) {
                    ++cnt;
                    ++v;
                }
                max_len = max(max_len, cnt);
            }
        }
        return max_len;
    }
};